app/models/card.rb

```diff
@@ -1,14 +1,14 @@
 class Card < ActiveRecord::Base
   validates :original_text, :translated_text, presence: true
-  validate :change_string
+  validate :change_string, if: proc { |a| a.original_text? && a.translated_text? }
 
   before_create :recheck_date
 
   scope :ready_for_review, -> { where("review_date <= ?", Time.now) }
   scope :random, -> { order('RANDOM()') }
 
   def check_word(user_text)
-    original_text.casecmp(user_text.strip.downcase).zero?
+    original_text.strip.mb_chars.downcase.casecmp(user_text.strip.mb_chars.downcase).zero?
```

Comment:
this line looks same with line 21, maybe you can rename this method and use it in validation?